meta-nilrt: update layer for compatibility with Yocto upstream master

shubhanshumt26 opened the pull request:
### Summary of Changes

This PR updates meta-nilrt to restore compatibility with the latest
Yocto upstream master branch. Recent upstream changes introduced
several build failures and deprecated behaviors that required fixes
across multiple recipes, patches, and layer integrations.

**The updates in this PR include:**

**1. Required adjustments for new upstream build behavior**

- Remove obsolete S assignments from git recipes
- Migrate source path assumptions from ${WORKDIR} to ${UNPACKDIR}
- Correct bbappend filenames to match updated upstream versions
- Update layer.conf to reflect current Yocto compatibility level
- Remove recipes that were dropped or deprecated in upstream layers

**2. Fixes for recipes broken by upstream changes**

- glibc-tests and glibc-locale-tests: correct LDFLAGS placement to ensure GNU_HASH and pass QA
- dkms: remove autotools-brokensep and adjust build/install steps
- libfmi: remove obsolete expat patch and update SRCREV
- xserver: unset VIRTUAL-RUNTIME_xserver_common to avoid conflicts
- mesa: update r600 enablement to match upstream AMD driver model
- llvm: enable AMDGPU and SPIRV targets at the distro level to satisfy
      Mesa radeonsi and mesa-clc build requirements after upstream AMD
      driver refactoring
- crda: add missing include workaround
- ni-grpc-device: fix shutdown symbol conflict

**3. Patch refreshes required due to upstream code drift**

- bolt patches refreshed
- xinput-calibrator desktop-file patch refreshed
- onboard patch regenerated and build fix added

### Justification

AB#[3039673](https://dev.azure.com/ni/DevCentral/_workitems/edit/3039673)
AB#[3039672](https://dev.azure.com/ni/DevCentral/_workitems/edit/3039672)


### Testing

* [x] I have built the core package feed with this PR in place. (`bitbake packagefeed-ni-core`)
* [x] I have built the core packagegroup desirable with this PR in place. (`bitbake packagegroup-ni-desirable`)
* [x] I have built the safemode-image with this PR in place. (`bitbake nilrt-safemode-rootfs`)
* [x] I have built the base-system-image with this PR in place. (`bitbake nilrt-base-system-image`)
* [x] I have built the nilrt-recovery-media with this PR in place. (`bitbake nilrt-recovery-media`)
* Additional testing performed:
Installed the generated images on a virtual machine.  Successfully booted into:
`nilrt-recovery-media`
`nilrt-base-system-image`
`nilrt-safemode-rootfs`
Verified system reaches expected boot state without errors.

### Procedure

* [x] I certify that the contents of this pull request complies with the [Developer Certificate of Origin](https://github.com/ni/nilrt/blob/HEAD/docs/CONTRIBUTING.md#developer-certificate-of-origin-dco).

shubhanshumt26 committed:
meta-nilrt: Remove obsolete S assignments in git recipes

Recent oe-core changes automatically set S for recipes
using Git as the source fetcher. Explicit overrides such as:

  S = "${WORKDIR}/git"
  S = "${UNPACKDIR}/git"

now trigger fatal errors during parsing:

This commit removes such instances across meta-nilrt to
restore compatibility with updated oe-core behavior.

Signed-off-by: Shubhanshu Mani Tripathi <shubhanshu.mani.tripathi@emerson.com>

shubhanshumt26 committed:
meta-nilrt: Update source directory assumptions to use ${UNPACKDIR}

oe-core now unpacks source into ${UNPACKDIR} instead of ${WORKDIR}.
Update recipes that previously assumed sources in ${WORKDIR} by:

 - Replacing S = "${WORKDIR}" with S = "${UNPACKDIR}"
 - Updating do_install and related steps to reference ${UNPACKDIR}
   or ${S}

This ensures correct file paths during build, install,
and packaging.

Signed-off-by: Shubhanshu Mani Tripathi <shubhanshu.mani.tripathi@emerson.com>

shubhanshumt26 committed:
meta-nilrt: Fix bbappend filenames to match updated upstream
recipe versions

BitBake was reporting missing .bbappend associations for:

 - mesa_%.bbappend
 - xinput-calibrator_git.bbappend
 - strongswan_5.%.bbappend

The upstream recipe names/versions have changed, so update the
append filenames accordingly to ensure they apply correctly:

 - mesa_%.bbappend → mesa.bbappend
 - xinput-calibrator_git.bbappend → xinput-calibrator_%.bbappend
 - strongswan_5.%.bbappend → strongswan_6.%.bbappend

These changes restore layer compatibility and ensure that the
bbappends are picked up by BitBake again.

Signed-off-by: Shubhanshu Mani Tripathi <shubhanshu.mani.tripathi@emerson.com>

shubhanshumt26 committed:
libfmi: drop obsolete static-expat patch and update SRCREV

Remove the temporary patch 0001-Remove-expat-static-linking.patch,
as the change has already been upstreamed.
Update SRCREV to the latest commit on the upstream master branch
so the recipe tracks current upstream sources without redundant
patching.
Rename the recipe file to reflect the current upstream release
version (3.0.4), since the previous filename still referenced the
older 3.0a4 release despite using a newer SRCREV.

Signed-off-by: Shubhanshu Mani Tripathi <shubhanshu.mani.tripathi@emerson.com>

shubhanshumt26 committed:
dkms: drop autotools-brokensep after autotools behavior change

dkms does not ship a configure script and is not autotools-based.
Previously, autotools.bbclass would skip oe_runconf when no
configure script was present, allowing this recipe to build.

Upstream autotools behavior changed to always run oe_runconf,
causing recipes without a configure script to fail:
  https://git.openembedded.org/openembedded-core/commit/?id=6d327a39befae44a88a812bdf4acde800dcee57b

As a result, inheriting autotools-brokensep is no longer valid
for dkms. Remove the inheritance, mark do_configure and
do_compile as noexec, and install explicitly via make install.

Extend FILES to include dkms_autoinstaller and common.postinst.

Signed-off-by: Shubhanshu Mani Tripathi <shubhanshu.mani.tripathi@emerson.com>

shubhanshumt26 committed:
ni-grpc-device: rename shutdown variable to avoid symbol conflict

The static variable `shutdown` was colliding with the `shutdown()`
function declared in <sys/socket.h>, leading to a compile-time
redefinition error when building core_server.cpp. To resolve this,
the variable has been renamed to `shutdown_server`, matching the
fix already applied upstream.

This patch is added temporarily to unblock the build until
ni-grpc-device is updated to the latest upstream revision, at
which point this change can be removed.

Upstream-Status: Inappropriate [Temporary workaround, will be dropped once upstream is updated]

Signed-off-by: Shubhanshu Mani Tripathi <shubhanshu.mani.tripathi@emerson.com>

shubhanshumt26 committed:
bolt: refresh patches to apply cleanly against updated source

Two existing patches failed with hunk offset/reject errors
due to code drift in the bolt sources. Both patches have
been refreshed and now apply cleanly:

 - 0001-daemon-support-running-without-PolicyKit.patch
 - 0002-exported-authorize-all-methods-by-default.patch

No functional changes were made to the patch contents themselves
— only context adjustments to align with the latest upstream
source layout.

Signed-off-by: Shubhanshu Mani Tripathi <shubhanshu.mani.tripathi@emerson.com>

shubhanshumt26 committed:
layer.conf: Set layer compatibility to whinlatter

Signed-off-by: Shubhanshu Mani Tripathi <shubhanshu.mani.tripathi@emerson.com>

shubhanshumt26 committed:
glibc-tests: fix missing GNU_HASH in ptest binaries

ptest binaries were missing GNU_HASH because LDFLAGS were added via
"CC += ${LDFLAGS}", which does not guarantee correct linker ordering.

Append LDFLAGS at the end of the compile commands so the linker
applies them correctly, fixing QA ldflags errors.

This change applies to:
- glibc-tests.bb
- glibc-locale-tests.bb

Signed-off-by: Shubhanshu Mani Tripathi <shubhanshu.mani.tripathi@emerson.com>

shubhanshumt26 committed:
meta-nilrt: remove xf86-input-vmmouse, pango-ptest, ndg-httpsclient

Drop these recipes since they are removed in upstream layers:

 - xf86-input-vmmouse, xf86-input-mouse: drop recipes -> https://github.com/openembedded/openembedded-core/commit/7c86172b7353b17ffbf8f277c1b8cc082a73c6a9
 - pango: upgrade 1.52.2 -> 1.54.0 ->  https://github.com/openembedded/openembedded-core/commit/228c4275df766d97250cbfa0a2c1254e704c07c7
 - python3-ndg-httpsclient: remove unused recipe -> https://github.com/openembedded/openembedded-core/commit/48b6851420ac54b181647bf23fe1ad86c75fa650

This aligns meta-nilrt with upstream and avoids maintaining obsolete
recipes.

Signed-off-by: Shubhanshu Mani Tripathi <shubhanshu.mani.tripathi@emerson.com>

shubhanshumt26 committed:
xserver: unset VIRTUAL-RUNTIME_xserver_common to avoid conflicts

Upstream changed xserver-common to x11-common(xserver-common: remove obsolete recipe -> https://github.com/openembedded/meta-openembedded/commit/64f742a625c9efb3d3cf3d3849abefc619a4a4b8)
and later merged x11-common to xserver-nodm-init. (x11-common: Merge into xserver-nodm-init > https://github.com/openembedded/meta-openembedded/commit/ee153327b046fe8474e498f39ecc87e573bc8604)

Setting VIRTUAL-RUNTIME_xserver_common to xserver-nodm-init causes a
package conflict with xserver-xfce-init, which both depends on and
conflicts with that package.

To avoid this issue for now, unset the virtual runtime:

  VIRTUAL-RUNTIME_xserver_common = ""

Update xserver-xfce-init to use the virtual instead of the old fixed
dependency.

This prevents the conflict until a proper resolution is implemented.

Signed-off-by: Shubhanshu Mani Tripathi <shubhanshu.mani.tripathi@emerson.com>

shubhanshumt26 committed:
mesa: switch r600 enablement to AMD PACKAGECONFIG model

Upstream Mesa changes removed PACKAGECONFIG[r600] and moved all Radeon
Gallium drivers (r300, r600, radeonsi) under a unified
PACKAGECONFIG[amd] entry. Because of this refactor, enabling "r600"
directly is no longer supported in the recipe.

Update mesa.bbappend to use PACKAGECONFIG += "amd", which now
implicitly enables the r600_dri driver required by PXIe-88XX hardware.
This aligns our override with the new upstream driver-selection model.

Reference upstream change:
mesa: introduce amd PACKAGECONFIG (commit - https://github.com/openembedded/openembedded-core/commit/6a2170ea7722c214412c05a7b98eede41611bb64)

Signed-off-by: Shubhanshu Mani Tripathi <shubhanshu.mani.tripathi@emerson.com>

shubhanshumt26 committed:
xinput-calibrator: refresh patch for desktop-file updates

The patch 'add-xfce-settings-and-wrapper-to-desktop-file.patch'
failed to apply after upgrading xinput-calibrator to version 0.8.0.
The upstream desktop file changed, causing a hunk failure in:

  scripts/xinput_calibrator.desktop

Regenerate the patch so it applies cleanly against the updated
desktop file while preserving the NI-specific additions.

Signed-off-by: Shubhanshu Mani Tripathi <shubhanshu.mani.tripathi@emerson.com>

shubhanshumt26 committed:
mesa: enable global opengl for mesa-native build

mesa-native only inherits global DISTRO_FEATURES and does not see
machine-specific overrides. Upstream Mesa now applies
ANY_OF_DISTRO_FEATURES globally, causing mesa-native to be skipped
unless opengl, opencl, or vulkan is enabled globally.

Append opengl to global DISTRO_FEATURES to avoid mesa-native being
skipped and breaking dependent builds.

Upstream ref: https://git.openembedded.org/openembedded-core/commit/?id=3daa991705c2c8e6e3d86a2c7e427f84176b31c2

Signed-off-by: Shubhanshu Mani Tripathi <shubhanshu.mani.tripathi@emerson.com>

rajendra-desai-ni commented on conf/distro/nilrt.conf:
```diff
@@ -14,7 +14,9 @@ DISTRO_FEATURES:append:x64 = "\
 
-VIRTUAL-RUNTIME_xserver_common = "xserver-common"
+# for now not setting it to xserver-nodm-init as that creates a package-conflict with xserver-xfce
+# that should be resolved as a separate bug.
```
Please create a bug to fix this or we might miss it in the future.

shubhanshumt26 replied:
Created a Bug for it and mentioned the details. FYI - [Bug](https://dev.azure.com/ni/DevCentral/_workitems/edit/3646392).

rajendra-desai-ni commented on recipes-support/salt/salt_3000.2.bb:
```diff
@@ -60,19 +58,19 @@ INITSCRIPT_PARAMS:${PN}-minion = "defaults 93 7"
+    install -m 0644 ${UNPACKDIR}/roster ${D}${sysconfdir}/${PN}/roster
     install -d ${D}${sysconfdir}/${PN}/cloud.conf.d ${D}${sysconfdir}/${PN}/cloud.profiles.d ${D}${sysconfdir}/${PN}/cloud.providers.d
 
     install -d ${D}${PYTHON_SITEPACKAGES_DIR}/${PN}-tests/
```
Below line has reference to S variable but you have removed the variable above. Hope you have verified the correct behavior?

shubhanshumt26 replied:
Already mentioned here: [Comment](https://github.com/ni/meta-nilrt/pull/908#discussion_r2621971196)

rajendra-desai-ni commented on recipes-connectivity/crda/crda_1.1.3.bb:
```diff
@@ -10,9 +10,9 @@ RDEPENDS:${PN} = "\
 	       udev \
 "
 
-S = "${WORKDIR}/git"
```
There are some S references in the file below, but the variable has been removed. Hope this change preserves previous behavior?

shubhanshumt26 replied:
`if s_dir_orig == '${WORKDIR}/git' or s_dir_orig == '${UNPACKDIR}/git':
        bb.fatal('Recipes that set S = "${WORKDIR}/git" or S = "${UNPACKDIR}/git" should remove that assignment, as S set by bitbake.conf in oe-core now works.')`
        
So, we don't need to set that.

rajendra-desai-ni commented on recipes-devtools/libfmi/libfmi_3.0.4.bb:
```diff
@@ -5,12 +5,9 @@ LICENSE = "BSD-3-Clause"
-SRCREV = "29523c20aec17277fc517900e3506d17d3f64642"
-
-S = "${WORKDIR}/git"
+SRCREV = "b711f5df98c8e341693679a6a08a891419a696d7"
```
You have updated the SRCREV to the latest hash in the master branch, but the file name still says 3.0a4 release. You may need to update the file name accordingly.
Or, can we pin the version to latest released version of libfmi which is 3.0.4 instead of pinning it to the latest hash in the master?

shubhanshumt26 replied:
Updated the filename with the latest released version.

rajendra-desai-ni commented on recipes-extended/lsb/lsbinitscripts_10.26.bb:
```diff
@@ -12,8 +12,6 @@ SRCREV = "79ec53026623b76d890b6dcdebf30d0c52d11a8b"
 UPSTREAM_CHECK_GITTAGREGEX = "^(?P<pver>\d+(\.\d+)+)"
 
 
-S = "${WORKDIR}/git"
```
This variable has been removed but is still being referenced in the file below. Hope it preserves expected behavior?

shubhanshumt26 replied:
Already mentioned here: [Comment](https://github.com/ni/meta-nilrt/pull/908#discussion_r2621971196)

rajendra-desai-ni commented on recipes-graphics/xinput-calibrator/xinput-calibrator/0001-add-xfce-settings-and-wrapper-to-desktop-file.patch:
```diff
@@ -12,17 +11,20 @@ done in a patch because it needs to be applied before xinput-calibrator
 --- a/scripts/xinput_calibrator.desktop
 +++ b/scripts/xinput_calibrator.desktop
-@@ -3,9 +3,9 @@ Name=Calibrate Touchscreen
- Name[de]=Touchscreen Kalibrieren
```
Can you confirm if this is the correct patch file changes? The changes seem a bit different compared to the [original patch file](https://github.com/ni/meta-nilrt/blob/nilrt/master/next/recipes-graphics/xinput-calibrator/xinput-calibrator/add-xfce-settings-and-wrapper-to-desktop-file.patch). The new patch shouldn't have modified these lines since they already exist in 0.8.0 version of xinput_calibrator package.

shubhanshumt26 replied:
Yes, this patch is correct.

The difference you’re seeing is due to upstream changes in the xinput-calibrator 0.8.0 .desktop file since the original NI patch was written in 2017.([FYI ](https://gitlab.freedesktop.org/xorg/app/xinput-calibrator/-/commit/ce0f1c8432270bffd64d5b6a117a68fd0e405e57))

The new patch preserves the exact same NI-specific behavior:

Switching the Exec entry to the NI wrapper script

Adding XFCE settings categories so the entry appears in the XFCE settings dialog

The additional Name[...] and Comment[...] entries were already present upstream and are not introduced or modified by this patch. The patch context changed, but the functional intent and effect remain identical to the original NI patch.

rajendra-desai-ni reviewed (changes requested):
- Follow 50-72 rule for the [commit](https://github.com/ni/meta-nilrt/pull/908/commits/a524ccd585367a6df32e0aed7578dcd7a7d9c725)
- Include details of all the testing you have done in the testing section
- Were you able to test the new BSI that includes your changes by installing on the VM/target?

shubhanshumt26 committed:
meta-nilrt: enable LLVM AMDGPU and SPIRV targets

Mesa radeonsi and mesa-clc require LLVM AMDGPU and SPIRV targets
to be available at build time. After the earlier change enabling
Mesa AMD support, LLVM must be configured accordingly at the
distro level to allow Mesa to configure successfully.

LLVM_TARGETS_TO_BUILD is a CMake list and must use semicolon
separators; space-separated values are treated as a single invalid
target and cause LLVM configuration failures.

This change complements the earlier commit in this series that
enables Mesa AMD support.
(mesa: switch r600 enablement to AMD PACKAGECONFIG model : ref commit - https://github.com/ni/meta-nilrt/pull/908/commits/a4b234a4fb2fbb04e70ed27f7cca797283f6b540)

Signed-off-by: Shubhanshu Mani Tripathi <shubhanshu.mani.tripathi@emerson.com>

chaitu236 commented on recipes-core/udev/udev-extraconf.bb:
```diff
@@ -38,7 +38,7 @@ do_install() {
 
 do_install:append:xilinx-zynq() {
-	install -m 0644 ${WORKDIR}/fpga.rules          ${D}${sysconfdir}/udev/rules.d/fpga.rules
+	install -m 0644 ${S}/fpga.rules          ${D}${sysconfdir}/udev/rules.d/fpga.rules
```
For all files: Why use S in some places and UNPACKDIR in other? Any reason not to stick with one?

shubhanshumt26 replied:
I tried to keep this consistent with how each recipe was already structured.
If a recipe was explicitly setting S before (earlier it was `S = "${WORKDIR}")`, I updated that to ` S = "${UNPACKDIR}"` and kept using `${S}` everywhere. That avoids changing the recipe structure more than necessary.

If a recipe never set `S`, I didn’t introduce it just for this change and used `${UNPACKDIR}` directly instead.
This matches current Yocto behavior where S is already handled by BitBake, and explicit overrides should only exist when the recipe actually needs them.

chaitu236 commented on recipes-kernel/dkms/dkms.bb:
```diff
@@ -18,15 +18,26 @@ SRC_URI = "\
-inherit autotools-brokensep
+# dkms does not ship a configure script and is not autotools-based.
+# Running standard oe_runconf/autotools tasks results in errors:
+#   ./configure: No such file
```
Apparently due to https://git.openembedded.org/openembedded-core/commit/meta/classes-recipe/autotools.bbclass?id=6d327a39befae44a88a812bdf4acde800dcee57b.

[optional] It isn't clear from commit description what changed that would cause this recipe to build on scarthgap but not dist-next.

shubhanshumt26 replied:
Updated the commit message accordingly.

chaitu236 commented on recipes-kernel/dkms/dkms.bb:
```diff
@@ -18,15 +18,26 @@ SRC_URI = "\
 
 
-FILES:${PN} += " ${datadir}/bash-completion/* ${datadir}/zsh/*"
+FILES:${PN} += " ${datadir}/bash-completion/* ${datadir}/zsh/* ${libdir}/dkms_autoinstaller ${libdir}/common.postinst"
```
Did you compare packages from this recipe on scarthgap vs dist-next to ensure everything remains same?

shubhanshumt26 replied:
Yes, I checked the scarthgap had these 5 packages for dkms.
<img width="864" height="99" alt="image" src="https://github.com/user-attachments/assets/1ab6b48d-a43e-4585-a8d2-390cf2c61b7a" />

Also, the dist-next has these 5 packages for dkms.
<img width="700" height="115" alt="image" src="https://github.com/user-attachments/assets/d8994117-aa56-493c-917d-b9a7d4c4f07e" />

chaitu236 commented on recipes-ni/ni-grpc-device/files/0001-Rename-shutdown-variable-to-shutdown_server-to-avoid.patch:
```diff
@@ -0,0 +1,50 @@
+but since the project has not yet been updated to the latest upstream revision,
+this patch temporarily introduces the same change locally to unblock the build.
+
+Upstream-Status: Inappropriate [Temporary patch which will be removed once ni-grpc-device is updated to latest upstream]
```
Do we have a PR up for ni-grpc-device? Or a WI to do that?

shubhanshumt26 replied:
The patch I’ve added here is temporary and will be dropped once we update the `ni-grpc-device` recipe to the latest upstream source, since this fix is already present there (FYI: [PR Link](https://github.com/ni/grpc-device/pull/1161)).

I did try upgrading the recipe, but that ran into additional, unrelated build errors. To avoid blocking this change, I’ve added the temporary patch for now.

shubhanshumt26 replied:
Also, created a [Tech Debt](https://dev.azure.com/ni/DevCentral/_workitems/edit/3659300) to remove this patch and linked it to update the meta-nilrt recipes WI.

chaitu236 commented on recipes-core/packagegroups/packagegroup-ni-runmode.bb:
```diff
@@ -74,7 +74,6 @@ RDEPENDS:${PN} = "\
 	python3-msgpack \
 	python3-multidict \
 	python3-multiprocessing \
-	python3-ndg-httpsclient \
```
FYI @AlexDanDuna in case System Link/salt is depending on this. This change goes in next major upgrade of NILRT (will probably ship in Q3/Q4)

chaitu236 commented on recipes-graphics/xinput-calibrator/xinput-calibrator/0001-add-xfce-settings-and-wrapper-to-desktop-file.patch:
```diff
@@ -12,17 +11,20 @@ done in a patch because it needs to be applied before xinput-calibrator
 
 Upstream-Status: Inappropriate [NI specific]
-    
-Signed-off-by: Alex Stewart <alex.stewart@ni.com>
```
Keep original sign off as well since those changes are still present. You can add a comment on next line on what you changed and add your own sign off tag after that.

e.g.,

```
Signed-off-by: Alex Stewart <alex.stewart@ni.com>
[Shubhanshu: Regenerated patch...]
Signed-off-by: Shubhanshu Mani Tripathi <shubhanshu.mani.tripathi@emerson.com>
```

shubhanshumt26 replied:
Updated!

chaitu236 commented on recipes-connectivity/crda/crda/0001-crda-add-missing-string.h-includes-in-reglib.c.patch:
```diff
@@ -0,0 +1,35 @@
+should be removed once CRDA is updated to a newer upstream release
+where the fix is already present.
+
+Upstream-Status: Inappropriate [temporary workaround; will be removed once recipe is updated]
```
What's the follow up WI#?

chaitu236 replied:
But better to backport the original patch https://github.com/mcgrof/crda/commit/6a7fd50cbb20a5495f54effd3f04c1387ed3fe28 instead of adding our own.

If we do that we don't need to have follow up WI for this; although we should have a WI to update at all these recipes that we have in meta-nilrt to latest upstream.

shubhanshumt26 replied:
Dropped this commit and pushed one new commit by backporting the original patch.
Also created a [WI](https://dev.azure.com/ni/DevCentral/_workitems/edit/3659174) to update meta-nilrt recipes.

chaitu236 commented on recipes-graphics/mesa/mesa.bbappend:
```diff
@@ -6,3 +6,7 @@
+
+# mesa-native is a native recipe and does not inherits machine specific overrides
+# so append opengl so that it builds.
+DISTRO_FEATURES:append = " opengl"
```
> mesa-native is a native recipe (runs on build host), and native recipes
only inherit "global" DISTRO_FEATURES, not MACHINE-specific overrides.

Why is this required? You've described what you're doing but not why.

> mesa-native is a native recipe and does not inherits machine specific overrides
 so append opengl so that it builds.

 When you say "so that it builds"; did you mean there was a build failure or somehow we need opengl during native builds? What upstream's master change caused the change in behavior from scarthgap?

shubhanshumt26 replied:
Updated everything in the commit with context.

chaitu236 commented on recipes-support/onboard/onboard_1.4.1.bb:
```diff
@@ -9,11 +9,12 @@ FILESEXTRAPATHS:prepend := "${THISDIR}/files:"
            file://NI.colors \
            file://NI.theme \
            file://onboard-defaults.conf \
+           file://0001-fix-compiler-compatibilities-issue.patch \
```
> regenerate autostart patch and add compiler fix

This commit is doing two things at once.

shubhanshumt26 replied:
I split it into two separate commits.

chaitu236 reviewed (changes requested):
>  I have built the core package feed with this PR in place. (bitbake packagefeed-ni-core)
 I have built the safemode-image with this PR in place. (bitbake nilrt-safemode-rootfs)
 I have built the base-system-image with this PR in place. (bitbake nilrt-base-system-image)
 I have built the nilrt-recovery-media with this PR in place. (bitbake nilrt-recovery-media)
 
 What about desirables packagegroup? Is it buildable or will it be addressed in another PR?

shubhanshumt26 committed:
crda: backport upstream fix for missing <string.h> includes

Backport upstream commit https://github.com/mcgrof/crda/commit/6a7fd50cbb20a5495f54effd3f04c1387ed3fe28 to fix build failures
caused by implicit declarations of memset() and memcmp() in
reglib.c when building with newer toolchains.

This backport can be removed once the CRDA recipe is updated to a
revision that already includes this fix.

Signed-off-by: Shubhanshu Mani Tripathi <shubhanshu.mani.tripathi@emerson.com>

shubhanshumt26 committed:
onboard: regenerate autostart patch for XFCE

Regenerate the autostart patch to update the desktop file and add
XFCE to the OnlyShowIn field. This ensures the onboard keyboard
autostarts correctly under the XFCE desktop environment.

The regenerated patch includes an updated description and proper
Upstream-Status metadata. Remove the old autostart patch reference
and add the refreshed patch to SRC_URI.

Signed-off-by: Shubhanshu Mani Tripathi <shubhanshu.mani.tripathi@emerson.com>

shubhanshumt26 committed:
onboard: fix compiler error by using stdbool.h

Modern toolchains fail to build onboard due to a legacy typedef
defining bool with enum values false and true, which conflicts
with standard C keywords.

Remove the custom typedef in osk_module.h and include <stdbool.h>
instead. This resolves build failures caused by redefinition of
'false' and 'true'.

Add the compiler fix as a separate patch to SRC_URI.

Signed-off-by: Shubhanshu Mani Tripathi <shubhanshu.mani.tripathi@emerson.com>

shubhanshumt26 commented:
> > I have built the core package feed with this PR in place. (bitbake packagefeed-ni-core)
> > I have built the safemode-image with this PR in place. (bitbake nilrt-safemode-rootfs)
> > I have built the base-system-image with this PR in place. (bitbake nilrt-base-system-image)
> > I have built the nilrt-recovery-media with this PR in place. (bitbake nilrt-recovery-media)
> 
> What about desirables packagegroup? Is it buildable or will it be addressed in another PR?

`packagegroup-ni-desirable` is passing(Updated PR Description).
But there were some issues in `packagefeed-ni-extra` that I have started looking. Will create a separate PR for that after fixing them.

chaitu236 commented on conf/distro/nilrt.conf:
```diff
@@ -14,7 +14,12 @@ DISTRO_FEATURES:append:x64 = "\
 
-VIRTUAL-RUNTIME_xserver_common = "xserver-common"
+# Mesa requires LLVM AMDGPU and SPIRV targets (CMake list, use semicolons).
+LLVM_TARGETS_TO_BUILD:append = ";AMDGPU;SPIRV"
```
[nit] Sorry didn't notice this in last round but shouldn't this be x64 specific (like [this](https://github.com/ni/meta-nilrt/blob/nilrt/master/scarthgap/recipes-core/images/nilrt-runmode-rootfs.bb#L13)) as it's graphics related (we don't support GUI on ARM targets)?

Doesn't need to be fixed in this PR - could be a follow up PR/WI.

shubhanshumt26 replied:
Ideally it should be x64-specific. I believe I tried that earlier but hit a build issue.
I’ll re-check, and if it works cleanly, I’ll address it in a follow-up PR.

shubhanshumt26 replied:
Created a [WI](https://dev.azure.com/ni/DevCentral/_backlogs/backlog/RTOS/Features%20and%20Risks?workitem=3685211) for it.
